docs/user-guides/search-config.md

```diff
@@ -0,0 +1,117 @@
+     The id should already exist in the NGT indexes before the search process.
+     The search algorithm is the same as `Search`.
+
+1. LinearSearch
+
+   - `LinearSearch` is the primary search algorithm with a query vector.
+     It searches all indexed vectors and calculates the distance between the query.
+     Its accuracy is exact, but the search time requires more than `Search` (ANN search) and increases the amount of indexed vector.
+
+1. LinearSearchById
+   - `LinearSearchById` is the primary search algorithm with the vector's id.
+     The id should already exist in the NGT indexes before the search process.
+     The search algorithm is the same as `LinearSearch`.
+
+<div class="notice">
+Linear Search service is available from Vald v1.4 or later.
+</div>
+
+For more details, please refer to [the Search API document](../api/search.md).
+
+## Configuration
+
+<!-- Describe Search parameters: ../../apis/proto/v1/payload -->
+<!-- TODO: divide search configuration for each agent algorithm -->
+
+Here is the current search config parameters.
+
+```bash
+// Represent search configuration.
+message Config {
+  // Unique request ID.
+  string request_id = 1;
+  // Maximum number of result to be returned.
+  uint32 num = 2 [ (validate.rules).uint32.gte = 1 ];
+  // Search radius.
+  float radius = 3;
+  // Search coefficient.
+  float epsilon = 4;
+  // Search timeout in nanoseconds.
+  int64 timeout = 5;
+  // Ingress filter configurations.
+  Filter.Config ingress_filters = 6;
+  // Egress filter configurations.
+  Filter.Config egress_filters = 7;
+  // Minimum number of result to be returned.
+  uint32 min_num = 8 [ (validate.rules).uint32.gte = 0 ];
+}
+```
+
+### request_id
+
+`request_id` is a unique request ID.
+It is **NOT** indexed vector's id.
+Users can use it for, e.g., the error handling process.
+
+### num
+
+`num` is the maximum number of search results you'd like to get.
+`num` should be a positive integer.
+
+### radius
+
+`radius`, the specific parameter for NGT, specifies the search range centered on the query vector in terms of the radius of a circle.
+The number of search target vectors increases along with the radius is large.
+There is a trade-off between accuracy and search speed.
+In many cases, it is hard to set it depending on dataset.
+
+The default value is the infinite circle.
+When setting a negative number as `radius`, `NGT` applies the radius as an infinite circle.
+
+<div class="notice">
+NGT will self-update the radius during the search process.
+</div>
+
+### epsilon
+
+`epsilon`, the specific parameter for NGT, specifies the search range's magnification coefficient (epsilon).
+NGT will use `(1+epsilon)*radius` as the search range.
+The number of search target vectors increases along with the epsilon being large.
+
+
+The default value (recommend value) is 0.1.
+While it is desirable to adjust this value within 0 - 0.3, it can also set a negative value (over than -1).
```

Comment:
epsilon depends on dataset, so the representation that is recommend value may be dangerous.
```suggestion
The default value is 0.1, and it may work in most cases.
However, the appropriate value may vary depending on dataset.
While it is desirable to adjust this value within 0 - 0.3, it can also set a negative value (over than -1).
```